server/internal/server/k8sclient.go

```diff
@@ -0,0 +1,44 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// K8sClient is a wrapper for Kubernetes clientsets.
+type K8sClient struct {
+	coreClientset kubernetes.Interface
+}
+
+// NewK8sClient creates an K8sClient instance.
+func NewK8sClient() (*K8sClient, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("get kube config: %w", err)
+	}
+	coreClientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("new kubernetes clientset %s: %w", config, err)
+	}
+
+	k := K8sClient{
+		coreClientset: coreClientset,
+	}
+	return &k, nil
+}
+
+// CoreClientset returns the Core clientset.
+func (k *K8sClient) CoreClientset() kubernetes.Interface {
+	return k.coreClientset
+}
+
+// GetEndPoints get a Service.
+func (k *K8sClient) GetEndPoints(ctx context.Context, name, namespace string) (*apiv1.Endpoints, error) {
```

Comment:
not sure if endpoints are good ones, but we can just get pod ips. cc @Ladicle

Reply:
Also not sure this will come eventually, but I think we need to watch instead of listing periodically.

Reply:
Removed.